packages/cli/src/commands/join.ts

```diff
@@ -174,21 +194,64 @@ packageVersion: string
         tag['x-displayName'] = tag['x-displayName'] || tag.name;
         tag.name = entrypointTagName;
         joinedDef.tags.push(tag);
+
+        if (skipTagsCheck) {
+          potentialConflicts.tags.description[entrypointTagName] = [entrypoint];
+        }
       }
-      if (!joinedDef[xTagGroups][indexGroup][Tags].find((t: any) => t === entrypointTagName)) {
-        joinedDef[xTagGroups][indexGroup][Tags].push(entrypointTagName);
+      // If tag already exist do not overwvite tag proprs,
+      // instead check if description are different for potential conflicts warning
+      else if (skipTagsCheck && tag.hasOwnProperty('description')) {
+        const isTagDescriptionNotEqual = joinedDef.tags.find(
+          (item: any) => item.description !== tag.description,
+        );
+        potentialConflicts.tags.description[entrypointTagName] = [
+          ...(potentialConflicts.tags.description[entrypointTagName] || []),
+          ...(isTagDescriptionNotEqual ? [entrypoint] : []),
+        ];
       }
 
-      const doesEntrypointExist = !potentialConflicts.tags.all[entrypointTagName] || (
-        potentialConflicts.tags.all[entrypointTagName] &&
-        !potentialConflicts.tags.all[entrypointTagName].includes(entrypoint)
-      )
+      if (!skipTagsCheck) createXTagGroups(entrypointFilename);
+
+      if (!skipTagsCheck) populateXTagGroups(entrypointTagName, getIndexGroup(entrypointFilename));
+
+      const doesEntrypointExist =
+        !potentialConflicts.tags.all[entrypointTagName] ||
+        (potentialConflicts.tags.all[entrypointTagName] &&
+          !potentialConflicts.tags.all[entrypointTagName].includes(entrypoint));
       potentialConflicts.tags.all[entrypointTagName] = [
-        ...(potentialConflicts.tags.all[entrypointTagName] || []), ...(doesEntrypointExist ? [entrypoint] : [])
+        ...(potentialConflicts.tags.all[entrypointTagName] || []),
+        ...(!skipTagsCheck && doesEntrypointExist ? [entrypoint] : []),
       ];
     }
   }
 
+  function getIndexGroup(entrypointFilename: string): number {
+    return joinedDef[xTagGroups].findIndex((item: any) => item.name === entrypointFilename);
```

Comment:
Do we have a type for `xTagGroups` defined? Can we use it here?

Reply:
This is good point, unfortunately  we do not have such type yet